backend/Controllers/UserController.cs

```diff
@@ -38,24 +40,26 @@ public ActionResult<UserResponse> Create([FromHeader] string authorization, [Fro
                 return new UnauthorizedResult();
             }
 
+
+            (string username, string password) = AuthHelper.GetUsernameAndPassword(authorization);
+
+            var check = _authService.IsValidLoginInfo(username, password);
+
+            if (!check)
+            {
+                return new UnauthorizedResult();
+            }
+
             try
             {
-                (string username, string password) = AuthHelper.GetUsernameAndPassword(authorization);
-
-                var check = _authService.IsValidLoginInfo(username, password);
-                if (!check)
-                {
-                    return new UnauthorizedResult();
-                }
-                
                 User createdUser = _userService.Create(newUserRequest);
-
-                return new UserResponse(newUserRequest);  
             }
-            catch (Exception)
+            catch (DuplicateUsernameException ex)
             {
-                return new UnauthorizedResult();
+                throw new Exception("Username already exists", ex);
```

Comment:
We actually want to catch the exception and _not_ throw a new one, instead returning a descriptive 400 Bad Request
```suggestion
                return BadRequest("Username already exists");
```